src/components/backgroundselector/BackgroundSelectorDirective.js

```diff
@@ -95,11 +97,24 @@ goog.require('ga_topic_service');
             dereg();
           });
 
-          scope.$on('gaTopicChange', function(event, newTopic) {
-            // If the layers config is loaded
-            if (gaLayers.getBackgroundLayers()) {
-              updateBgLayer(newTopic);
-            }
+          gaTopic.getTopics().then(function() {
+            scope.$on('gaLayersChange', function() {
```

Comment:
> We don't have garantueed order of exection between the getTopics promise and the gaLayersChange event

Indeed. The change I proposed requires work not to encounter #2520 again.

> Also, gaLayersChange will be called on every language change, not only once at startup. But it won't get called on topic change!

So this correction is very incomplete.

> What's the difference between the gaLayersChange part here and the one above?

From what I understand from your code, here we should react on topic change to force the change of background layers when the topic change. The one above is there to use the background layer defined in the permalink once the layers are loaded. I just added the `gaLayersChange` event to be "sure" that the background layers are loaded once we have the topic.

How can we make this work then? To populate `defaultBgOrder` I need:
- the topic which defines which layers are background layer
- the layers

I think that for this to work properly, we must: on `gaTopicChange` wait for the new promise which gets the topic to resolve. In this promise wait for a promise which gets the layers to resolve. This implies to have a new promise for the layers. This is what I intended to do at first before being lazy and just using the existing `gaLayersChange`. Let the reaction to `gaLayersChange` untounched.

What do you think?